plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/TreeItemHandler.java

```diff
@@ -381,6 +382,60 @@ public void run() {
 			}
 		});
 	}
+	/**
+	 * Clicks on specified TreeItem
+	 * @param swtTreeItem
+	 */
+	public void click(final TreeItem swtTreeItem) {
+		Rectangle bounds = getBounds(swtTreeItem);
+		notifyMouseClick(swtTreeItem,bounds.x + (bounds.width / 2), bounds.y + (bounds.height / 2));
+	}
+	
+	/**
+	 * Get bounds of specified TreeItem bounds. TreeItem should be enabled before calling this method.
+	 * @param swtTreeItem
+	 * @return tree item bounds
+	 */
+	public Rectangle getBounds(final TreeItem swtTreeItem) {
+		return Display.syncExec(new ResultRunnable<Rectangle>() {
+			@Override
+			public Rectangle run() {
+				return swtTreeItem.getBounds();
+			}
+		});
+	}
+	/**
+	 * Notifies specified TreeItem about Mouse Click over it
+	 * @param swtTreeItem
+	 * @param x
+	 * @param y
+	 */
+	private void notifyMouseClick(TreeItem swtTreeItem, int x, int y) {
+		TreeHandler.getInstance().notifyTree(swtTreeItem, SWT.MouseDown, createMouseEvent(swtTreeItem, x, y, 1, SWT.NONE, 1));
+		TreeHandler.getInstance().notifyTree(swtTreeItem, SWT.MouseUp, createMouseEvent(swtTreeItem,x, y, 1, SWT.BUTTON1, 1));
+	}
+	/**
+	 * Creates Mouse Event for TreeItem
+	 * @param swtTreeItem
+	 * @param x
+	 * @param y
+	 * @param button
+	 * @param stateMask
+	 * @param count
+	 * @return
+	 */
+	private Event createMouseEvent(TreeItem swtTreeItem, int x, int y, int button, int stateMask, int count) {
```

Comment:
I didn't notice that event.widget has to be treeItem. Then add event.item and event.type to this method